src/main/java/com/google/firebase/messaging/FcmOptions.java

```diff
@@ -0,0 +1,68 @@
+package com.google.firebase.messaging;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.api.client.util.Key;
+import com.google.firebase.internal.FcmOptionsUtil;
+
+/**
+ * Represents the platform-independent FCM options that can be included in a {@link Message}.
+ * Instances of this class are thread-safe and immutable.
+ */
+public class FcmOptions {
+
+  @Key("analytics_label")
+  private final String analyticsLabel;
+
+  private FcmOptions(Builder builder) {
+    checkArgument(FcmOptionsUtil.validAnalyticsLabelFormat(builder.analyticsLabel),
+        "Analytics label must have format matching'^[a-zA-Z0-9-_.~%]{1,50}$'");
```

Comment:
I see this error message is repeated in several places. You can avoid this duplication by slightly modifying your util method:

```
public static void checkAnalyticsLabel(String label) {
  checkArgument(isValid(label), "Analytics label must have format matching'^[a-zA-Z0-9-_.~%]{1,50}$");
}

// Then you can call this as...
FcmOptionsUtil.checkAnalyticsLabel(label);
```

Reply:
done.